### PR TITLE
CPPC Service group improvements

### PR DESCRIPTION
**Pull request by pathakraul**
Remove the Hard ID parameter from Fastchannel region request
Convert functions names to monospace in rpmi mpxy chapter
Add Perf Feedback Fastchannel in CPPC ServiceGroup

### CODE REVIEW

File: src/srvgrp-cppc.adoc [hunk 1]
```diff
@@ -77,21 +77,34 @@ The following table lists the services in the CPPC service group:
...
+message format. The simple data format supported by the fast-channel allows
+faster processing of the performance change requests made through a fast-channel
+and faster read of the performance feedback values supported over the
+fast-channel.
+
+The CPPC service group defines two types of fast-channel for each application
+processor.
 
-If the fast-channels are implemented, the application processor will either write
+===== Performance Request fast-channel.
```
Comment (lftan) [inline src/srvgrp-cppc.adoc:93]: "Fast-channel", and remove full stop for text header.

File: src/srvgrp-cppc.adoc [hunk 2]
```diff
@@ -122,33 +135,58 @@ Otherwise the application processor can call the service `CPPC_WRITE_REG` for th
 ! 0x4
 ! Maximum performance level
 !===
+|===
+
+===== Performance Feedback fast-channel.
```
Comment (lftan) [inline src/srvgrp-cppc.adoc:140]: "Fast-channel", and remove full stop for text header.

File: src/srvgrp-cppc.adoc [hunk 3]
```diff
@@ -77,21 +77,34 @@ The following table lists the services in the CPPC service group:
...
-Otherwise the application processor can call the service `CPPC_WRITE_REG` for the
-`DesiredPerformanceRegister` or `MinimumPerformanceRegister` and
-`MaximumPerformanceRegister`.
+and maximum performance level in case of Autonomous (CPPC2) mode in the
+fast-channel. Otherwise the application processor can call the service
+`CPPC_WRITE_REG` for the `DesiredPerformanceRegister` or
+`MinimumPerformanceRegister` and `MaximumPerformanceRegister`.
+
+The layout of this fast-channel which depends on the CPPC mode, either normal or
+autonomous mode is discoverable through a defined service.
```
Comment (lftan) [inline src/srvgrp-cppc.adoc:102]: Maybe can change "defined service" to a service name is better.

File: src/srvgrp-cppc.adoc [hunk 4]
```diff
@@ -77,21 +77,34 @@ The following table lists the services in the CPPC service group:
...
 
-[#table_cppc_fastchan]
-.CPPC Fast-channel Layout
+The size of this fast-channel type is of 8-byte.
```
Comment (lftan) [inline src/srvgrp-cppc.adoc:104]: The size of this fast-channel type is 8 bytes.

File: src/srvgrp-cppc.adoc [hunk 5]
```diff
@@ -122,33 +135,58 @@ Otherwise the application processor can call the service `CPPC_WRITE_REG` for th
...
+
+===== Performance Feedback fast-channel.
+In this fast-channel the application processor will read the 
+supported value for estimating the delivered performance as performance feedback
+for an application processor. The application processor current frequency (Hz)
+is used for performance feedback in this fast-channel. The platform
+microcontroller must write the frequency of an application processor in the
+fast-channel whenever it changes.
 
+The size of this fast-channel type is of 8-byte.
```
Comment (lftan) [inline src/srvgrp-cppc.adoc:148]: The size of this fast-channel type is 8 bytes.

File: src/srvgrp-cppc.adoc [hunk 6]
```diff
@@ -122,33 +135,58 @@ Otherwise the application processor can call the service `CPPC_WRITE_REG` for th
...
-fast-channel. The `base-address` and `size` (bytes) of the fast-channels shared
-memory region can be discovered using the service `CPPC_GET_FAST_CHANNEL_REGION`.
-The layout of each fast-channel which depends on the CPPC mode, either normal or
-autonomous mode can also be discovered through the same service.
-
-A fast-channel size for a application processor is of `8-byte` with two fields
-of `4-byte` each. A fast-channel must be naturally aligned to the size of the
-fast-channel.
+| 0x0
+| Current frequency Low 32-bit (Hz)
```
Comment (lftan) [inline src/srvgrp-cppc.adoc:158]: low

File: src/srvgrp-cppc.adoc [hunk 7]
```diff
@@ -122,33 +135,58 @@ Otherwise the application processor can call the service `CPPC_WRITE_REG` for th
...
 
-The offset of a fast-channel entry belonging to a application processor in that
-shared memory region can be discovered through service `CPPC_GET_FAST_CHANNEL_OFFSET`.
-The offset can be added into the `base-address` of the shared memory region to
-form the address of a application processor fast-channel.
+| 0x4
+| Current frequency High 32-bit (Hz)
```
Comment (lftan) [inline src/srvgrp-cppc.adoc:161]: high

File: src/srvgrp-cppc.adoc [hunk 8]
```diff
@@ -122,33 +135,58 @@ Otherwise the application processor can call the service `CPPC_WRITE_REG` for th
...
+(bytes) of the fast-channels shared memory region can be discovered using the
+service `CPPC_GET_FAST_CHANNEL_REGION`. The `base-address` of the shared memory
+region must be aligned to 8-byte which is maximum size of a fast-channel in
+both the types.
+
+The offsets of fast-channel of both types for an application processor are aligned
+to 8-byte. The offset of both fast-channel types in the shared memory region can
+be discovered through service `CPPC_GET_FAST_CHANNEL_OFFSET`. The offsets
+discovered can be added into the `base-address` of the shared memory region to
+form the address of Performance Request fast-channel and Performance Feedback
```
Comment (lftan) [inline src/srvgrp-cppc.adoc:181]: "Performance Request Fast-channel"

File: src/srvgrp-cppc.adoc [hunk 9]
```diff
@@ -122,33 +135,58 @@ Otherwise the application processor can call the service `CPPC_WRITE_REG` for th
...
+fast-channel for an application processor.
```
Comment (lftan) [inline src/srvgrp-cppc.adoc:182]: "Performance Feedback Fast-channel"

File: src/srvgrp-cppc.adoc [hunk 10]
```diff
@@ -122,33 +135,58 @@ Otherwise the application processor can call the service `CPPC_WRITE_REG` for th
...
+
+===== Performance Request Fast-channel Doorbell
+A doorbell can also be supported for this fast-channel type which is shared
+between all the application processors. If a doorbell is available then it must
+be supported through a read-modify-write sequence to a memory-mapped register.
+The doorbell details and attributes can be discovered by the application
+processor through the defined service in this service group.
```
Comment (lftan) [inline src/srvgrp-cppc.adoc:189]: can change "defined service" to the service name

File: src/srvgrp-cppc.adoc [hunk 11]
```diff
@@ -587,14 +624,24 @@ processor in the shared memory region containing all the fast-channels.
 - Other errors <<table_error_codes>>
 
 | 1
-| OFFSET_LOW
+| PERF_REQUEST_OFFSET_LOW
 | uint32
-| Lower 32-bit of a fast-channel offset.
+| Lower 32-bit of a Performance Request fast-channel offset.
```
Comment (lftan) [inline src/srvgrp-cppc.adoc:629]: Should change "fast-channel" to "Fast-channel"? 
Assume this is a term.

Comment (pathakraul) [reply src/srvgrp-cppc.adoc:629]: In headings its ok, else where it looks odd with capital Fast-channel. So i have used "fast-channel" in the text and "Fast-channel" in headings

**Pull request merged**